[tools] versioning libhermes.so soname for android

Kudo opened the pull request:
# Why

for expo-go to load different hermes-engine version, versioning libhermes.so on filename is not enough, we should further update the `soname`. otherwise, android will use loaded soname directly.

# How

use [patchelf](https://github.com/NixOS/patchelf) to update soname in versioning stage.

# Test Plan

based on #13793 and this change, re-generate sdk-42 aar and build versioned android expo-go. load these two projects without close activity
 - load [sdk42 hermes bundle](https://expo.dev/@kudochien/sdk42) which hbc version is 74
 - load local unversioned hermes project based on react-native 0.64 (hermes-engine 0.7.2) which hbc version is 76

before the change, expo-go will crash.
after the change, expo-go could load both two projects without problems.


# Checklist

after the pr is merged, will further cherry-pick this to sdk-42 branch and re-generate aar.

Kudo committed:
[tools] versioning libhermes.so soname for android

tsapeta reviewed (commented):
Looks good, but it would be even better if we can somehow check if you have `patchelf` installed at the beginning of the versioning process 😉

Kudo committed:
check patchelf installation earlier

Kudo committed:
upgrade for rn 0.64 upgrade

after https://github.com/facebook/react-native/commit/be53aae324546ca038e2b58184ecc7066cfb598e,
`prepareHermes` is a Copy task. we should revert back as generic task to
support both copy and exec.
